### ISSUE TITLE
Inner produc is unintuitive as interpretation of "*" or "×" between vectors

### ISSUE THREAD

**Issue by schuelermine** [Qalculate/libqalculate#257]
Since `×` is interpreted the same as `*`, I believe `*` should be interpreted as the cross product between two vectors when both are vectors, and as the scalar product if one operand is a scalar. Instead, I suggest using `.` for the inner product (this is, for example, what WolframAlpha uses), and to convert `·` to `.` if needed.

**Comment by hanna-kn** [comment]
I agree that this is problematic. It is however complicated to find a good solution that works with the general multiplication operator usage. 

The two solutions I currently consider are the following.
1. Show a warning when two 3-dimensional vectors are multiplied and advice the user to use cross() or dot().
2. Remove support for vector products using standard multiplication operators (including implicit multiplication) and show an an error instead (advising the user to use an appropriate function instead). Possibly add support for dot and cross product using · and × when it is obvious that the factors are vectors.

Solution 1 has the benefit of backwards compatibility, but I think that 2 is otherwise a better solution.
